File: two_divination/no3.cpp
// TODO: Formulas: M=month, D=day, S=(M*2+D)%3
#include <iostream>
using namespace std;

int main() {
  int month, day;

  cin >> month >> day;
  if (month < 1 || month > 12 || day < 1 || day > 31) {
    cout << "Invalid input!" << endl;
    return 1;
  }

  int fortune = (month * 2 + day) % 3;

  switch (fortune) {
  case 2:
    cout << "大吉" << endl;
    break;
  case 1:
    cout << "吉" << endl;
    break;
  default:
    cout << "普通" << endl;
  }

  return 0;
}

File: two_divination/no3.java
// TODO: Formulas: M=month, D=day, S=(M*2+D)%3
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int month = input.nextInt();
        int day = input.nextInt();

        // Input validation
        if (month < 1 || month > 12 || day < 1 || day > 31) {
            System.out.println("Invalid input!");
            input.close();
            return;
        }

        int fortune = (month * 2 + day) % 3;
        switch (fortune) {
        case 2:
            System.out.println("大吉");
            break;
        case 1:
            System.out.println("吉");
            break;
        default:
            System.out.println("普通");
        }
        input.close();
    }
}
